pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -577,11 +577,48 @@ func (r *Reconciler) processTargetPools(desired bool, poolFunc poolProcessor) er
 	return nil
 }
 
+// ensureInstanceGroupExists checks whether an instancegroup exists.
+func (r *Reconciler) ensureInstanceGroupExists(instanceGroupName string) error {
+	_, err := r.computeService.InstanceGroupGet(r.projectID, r.providerSpec.Zone, instanceGroupName)
+	if err != nil {
+		return fmt.Errorf("instanceGroupGet request failed: %v", err)
+	}
+	return nil
+}
+
+// ensureInstanceGroupInBackendService checks whether an instancegroup is assigned to a backend service.
+func (r *Reconciler) ensureInstanceGroupInBackendService() error {
+	backendService, err := r.computeService.BackendServiceGet(r.projectID, r.providerSpec.Region, r.backendServiceName())
+	if err != nil {
+		return fmt.Errorf("backendServiceGet request failed: %v", err)
+	}
+
+	for _, backend := range backendService.Backends {
+		if backend.Group == r.FQDNInstanceGroup() {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("couldn't find the instance group %s in a backend service", r.FQDNInstanceGroup())
+}
+
 // registerInstanceToControlPlaneInstanceGroup ensures that the instance is assigned to the control plane instance group of its zone.
 func (r *Reconciler) registerInstanceToControlPlaneInstanceGroup() error {
 	instanceSelfLink := fmtInstanceSelfLink(r.projectID, r.providerSpec.Zone, r.machine.Name)
 	instanceGroupName := r.controlPlaneGroupName()
 
+	if err := r.ensureInstanceGroupExists(instanceGroupName); err != nil {
```

Comment:
Is there a way to distinguish between a not found error and any other error? 

Also, when we use `ensure` we mean it to be, check if the thing exists, create it if it doesn't. So you should move the `registerNewInstanceGroup` into the `ensureInstanceGroup` function